trace_context/HTTP_HEADER_FORMAT_RATIONALE.md

```diff
@@ -61,3 +63,21 @@ Lowercase names has a few benefits:
 
 Url encoding is low-overhead way to encode unicode characters for non-latin characters in the 
 values. Url encoding keeps a single words in latin unchanged and easy readable.
+
+## Versioning
+
+Versioning options are:
+
+1. Pass thru unknown headers
+2. Re-start trace when you see unknown header
+3. Try to parse trace following some rules when you see unknown header
+
+One variation is whether original or new header you cannot recognize is preserved in `tracestate`.
+
+- Option 1 is least favorable as it makes one bad header break entire distributed trace.
+- Option 2 is better. It's easy, doesn't restrict future version in any way and re-started trace should be understood by new systems. So only one "connection" is lost. And the lost connection issue can be solved by storing original header in `tracestate`. Drawbacks are also obvious. First, single old component always breaks traces. Second, it's harder to transition without customer dissat of broken traces. 
+
+  Storing original value also has negative effects. Valid `traceparent` is 55 characters (out of 512 allowed for `tracestate`). And "bad" headers could be much longer pushing valuable `tracestate` pairs out. Also this requirement increases the chance of abuse. When bad actor will start sending header with the version `99` that is only understood by that actor. And the fact that every system passes thru original value allows this actor to build complete solution based on this header.
```

Comment:
sorry maybe I missed something.. are you implying the data in 512 <= traceparent.length + tracestate.length? Ex in the variable length traceparent (which versioning implies how it is defined now) traceparent grows at the expense of tracestate?